src/main/java/com/garbagemule/MobArena/SpawnsPets.java

```diff
@@ -57,8 +58,13 @@ private void spawnPetsFor(Player player, Arena arena, Material material, EntityT
         int amount = inv.getItem(index).getAmount();
         for (int i = 0; i < amount; i++) {
             Entity pet = arena.getWorld().spawn(player.getLocation(), entity.getEntityClass());
-            pet.setCustomName(player.getDisplayName() + "'s pet");
-            pet.setCustomNameVisible(true);
+            if (!petName.isEmpty()) {
+                String formattedName = ChatColor.translateAlternateColorCodes('&',
+                                       petName.replace("<player-name>", player.getName())
+                                              .replace("<display-name>", player.getDisplayName()));
```

Comment:
Not a fan of this kind of alignment and nested expressions. I'd much rather just split it into a couple of statements:
```java
String resolved = petName
    .replace("<player-name>", player.getName())
    .replace("<display-name>", player.getDisplayName());
String colorized = ChatColor.translateAlternateColorCodes('&', resolved);
pet.setCustomName(colorized);
```
Note the `.replace()` chain starting on the next line, rather than on the same line as the variable. This is to stay consistent with what we do in the [sign template rendering](https://github.com/garbagemule/MobArena/blob/903752d23ae884e1a74e51079016bafbd414d0c6/src/main/java/com/garbagemule/MobArena/signs/RendersTemplate.java#L58-L86).

The variable names are a bit arbitrary, and there are probably better names for them. Naming things is always difficult, but I do think the splitting improves readability, so I think it's worth it.